include/data/nodeid.h

```diff
@@ -54,5 +54,9 @@ class NodeID {
   explicit operator bool() const;
 };
 
+struct NodeIDHash {
```

Comment:
https://google.github.io/styleguide/cppguide.html#std_hash
```
Do not define specializations of std::hash.
```
Seems this is discouraged.